src/Microsoft.Extensions.HashCodeCombiner.Sources/HashCodeCombiner.cs

```diff
@@ -70,6 +70,7 @@ public void Add(object o)
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Add<TValue>(TValue value, IEqualityComparer<TValue> comparer)
+            where TValue : class
```

Comment:
Hmm, you're right. Was because the null equality test made no sense for a struct; but then it doesn't really matter either; so will revert.